docs/core/concepts/results.md

```diff
@@ -59,33 +50,53 @@ The most common scenario in which a user might need to directly interact with a
 <Result: 1>
 ```
 
-## Result Handlers
+## Persisting Results
+
+Subclasses of `Result` that align with a storage backend have a specific implementation of a `read` / `write` / `exists` interface for persisting its data. These can be used directly by a user to read, write, or check on arbitrary data in storage backends matching the result subclass. They are also used by the Prefect pipeline whenever checkpointing is turned on if it needs to retrieve results from persistent storage.
 
-Result handlers are a more public entity than the `Result` class, as they provide the means to persist the data from an in-memory `Result` object. A result handler is a specific implementation of a `read` / `write` interface for handling data. The only actual requirement for a result handler implementation is that the `write` method returns a JSON-compatible object. 
+Each of these methods depend on the `Result.location` attribute to determine where the data should actually be persisted. This `location` attribute is the only value serialized to the database of Prefect Core's server or Prefect Cloud. By default this location is autogenerated as a UUID, but it can be provided as a formattable string by the user to configure exactly where the result is located.
 
-In general, however, result handler `write` methods are implemented so that they do two things:
-1. serialize the return value of a Task in-memory in the Task State's `Result.value` attribute, and persist it to disk in the result handler's storage backend, and
-2. segregate out a "safe value", usually metadata, about that stored return value in the State's `Result.safe_value` attribute and, for Cloud customers, persist that safe value in the scheduler database.
+### Persisting user-created `Result`s
+If you want to use the `Result` API yourself, such as in a task, instantiate a configured result object and use its `read`, `write`, and `exists` methods directly however you want to in your task.
 
-On the flip side, result handler `read` methods usually access the metadata in an upstream `Result.safe_value` to locate, and then deserialize, the upstream `Task`'s return value that is in storage.
+For example, if your task needs to read a file from an S3 bucket, consider using Prefect's `S3Result` to identify the S3 file and read it:
 
-For example, we can easily imagine different kinds of result handlers:
+```python
+from prefect import Flow, task
+from prefect.engine.results import S3Result
+
+@task
+def my_task():
+    my_file_in_S3 = S3Result(bucket='mybucket', location='myfile.txt')
+    contents = my_file_in_S3.read()
```

Comment:
At this moment a `location` is required in the signature of `read`

Reply:
Fixed and I think overall better in e315dd32d8955b6bb109411ad77c6191ce12af78